### ISSUE TITLE
[Select] Select isn't visible if all options are disabled

### ISSUE THREAD

**Issue by whalesync-ryder** [radix-ui/primitives#1663]
### Current Behavior

Create a Select containing only items that are disabled.
When you open the select, the menu shows up at the bottom of the document, or even not visible (in my prod code with a more complicated dom).

Example:
![Screen Shot 2022-09-14 at 4 07 07 PM](https://user-images.githubusercontent.com/93688278/190278526-4e63010e-22d4-4860-b8f9-ad22e8e95e30.png)

### Expected behavior
The menu should open on top of the select. Even though there's nothing that can be selected, it should still open in the same location.

### Reproducible example

[Demo](https://codesandbox.io/s/radix-select-all-disabled-repro-32t2hw)

Relevant code: A simple select with only disabled values:
```
 <Select.Root>
            <Select.Trigger>
              <Select.Value>My trigger</Select.Value>
              <Select.Icon />
            </Select.Trigger>
            <Select.Portal>
              <Select.Content>
                <Select.Viewport>
                  <Select.Item value="france" disabled>
                    <Select.ItemText>France</Select.ItemText>
                  </Select.Item>
                  <Select.Item value="united-kingdom" disabled>
                    <Select.ItemText>United Kingdom</Select.ItemText>
                  </Select.Item>
                  <Select.Item value="spain" disabled>
                    <Select.ItemText>Spain</Select.ItemText>
                  </Select.Item>
                </Select.Viewport>
              </Select.Content>
            </Select.Portal>
          </Select.Root>
```

### Suggested solution

I'm guessing the placement logic that looks for the first item doesn't do well if there's no first item.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |@radix-ui/react-select         |    1.0.0     |
| React            | n/a     |    17.0.1     |
| Browser          | Chrome        |    105.0.5195.52     |
| Assistive tech   |    n/a     |   n/a      |
| Node             | n/a     |    v16.13.0     |
| npm/yarn         |    yarn     |         |
| Operating System |    MacOS     |   12.3.1      |

**Comment by benoitgrelard** [comment]
Hey @whalesync-ryder, thanks for reporting this.

We have some logic to help aligning in case there is no `value` or `defaultValue` (ie. no current selection), it will default to the first valid item (ie. the first non-disabled item).

We'll see if we can improve this but that's a pretty unusual scenario as it will only happen if:
- you have no `value` or `defaultValue` (ie. no current selection)
- you have disabled all of the options one by one

That's pretty unusual because if you no option is selectable, a better UX would be to disable the entire select anyway by passing `disabled` to the `Trigger` part.

**Comment by whalesync-ryder** [comment]
Thanks for looking at it.

I don't really agree with your last statement. There's different semantics to "this select is disabled" and "this select only contains unselectable items".  The first usually means "this whole select doesn't apply right now", whereas the second one is about the state of the individual items.

In my scenario, for example, an item is disabled in the select if it isn't "compatible" with the user's current configuration. It's important for the user to open up the menu to see that everything is disabled, so they can re-use the same visual language for "incompatible" items, whether half the items are "incompatible" or all of them are.  If I disable the entire select, that's a new concept for the user and I need to add another piece of UI to explain why it's broken. 

Thanks again.

**Comment by benoitgrelard** [comment]
> Thanks for looking at it.
> 
> I don't really agree with your last statement. There's different semantics to "this select is disabled" and "this select only contains unselectable items". The first usually means "this whole select doesn't apply right now", whereas the second one is about the state of the individual items.
> 
> In my scenario, for example, an item is disabled in the select if it isn't "compatible" with the user's current configuration. It's important for the user to open up the menu to see that everything is disabled, so they can re-use the same visual language for "incompatible" items, whether half the items are "incompatible" or all of them are. If I disable the entire select, that's a new concept for the user and I need to add another piece of UI to explain why it's broken.
> 
> Thanks again.

Yeah that's a good point about expectations. Although one could argue having to click to "discover" there is nothing available perfect either.

Regardless, I think your request is valid so we'll look into it 🙂

**Comment by iamterryclark** [comment]
We are having a similar issue in that all of our options are disabled, due to different user machines.

Our example uses dropdown for GPU supporting hardware acceleration, and to select which GPU is valid for use.
 
We would like to display all possible GPUs (integrated or dedicated) and if they both do not support hardware acceleration, they will be disabled. Next to these options we wanted to insert an icon with a tooltip in order to allow the user to find out why they are not selectable.

So this is another exception i believe.

In order to achieve this we may need to have our own disabled styling (different from the Radix default disabled option)

I have also seen at times that the dropdown half of the component will display elsewhere on the page (similar to the above) and not directly underneath the select area. Also tooltips will not work with disabled items as it disables mouse events as well.
